math/dec.go

```diff
@@ -459,38 +459,24 @@ func (d LegacyDec) ApproxRoot(root uint64) (guess LegacyDec, err error) {
 		return absRoot.NegMut(), err
 	}
 
-	// One decimal, that we invalidate later. Helps us save a heap allocation.
-	scratchOneDec := LegacyOneDec()
-	if root == 1 || d.IsZero() || d.Equal(scratchOneDec) {
+	if root == 1 || d.IsZero() || d.Equal(LegacyOneDec()) {
 		return d, nil
 	}
 
 	if root == 0 {
-		return scratchOneDec, nil
+		return LegacyOneDec(), nil
 	}
 
-	guess, delta := scratchOneDec, LegacyOneDec()
-	smallestDec := LegacySmallestDec()
+	guess, delta := LegacyOneDec(), LegacyOneDec()
 
-	for iter := 0; delta.AbsMut().GT(smallestDec) && iter < maxApproxRootIterations; iter++ {
-		// Set prev = guess^{root - 1}, with an optimization for sqrt
-		// where root=2 => prev = guess. (And thus no extra heap allocations)
-		prev := guess
-		if root != 2 {
-			prev = guess.Power(root - 1)
-		}
+	for iter := 0; delta.Abs().GT(LegacySmallestDec()) && iter < maxApproxRootIterations; iter++ {
```

Comment:
We can start by cheaper comparisons firstly of just integers then onto the .Abs().GT so
```go
for iter := 0; iter < maxApproxRootIterations && delta.Abs().GT(legacySmallestDec); iter++ {
```

and even more `LegacySmallestDec()` never changes so we can initiate it as a global so
```
var legacySmallestDec = LegacySmallestDec()
```

Reply:
[f4dff23](https://github.com/cosmos/cosmos-sdk/pull/17725/commits/f4dff23a123556ded726c75668f02cd206905c40)